testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -134,6 +141,131 @@ public void testColumnMappingMode(String mode)
         }
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91,
+            PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingWithTrueAndFalseDataProvider")
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testColumnMappingModeAllDataTypes(String mode, boolean partitioned)
+    {
+        String tableName = "test_dl_column_mapping_mode_name_all_types_" + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName + " (" +
+                "    a_boolean BOOLEAN," +
+                "    a_tinyint TINYINT," +
+                "    a_smallint SMALLINT," +
+                "    a_int INT," +
+                "    a_bigint BIGINT," +
+                "    a_decimal_5_2 DECIMAL(5,2)," +
+                "    a_decimal_21_3 DECIMAL(21,3)," +
+                "    a_double DOUBLE," +
+                "    a_float FLOAT," +
+                "    a_string STRING," +
+                "    a_date DATE," +
+                "    a_timestamp TIMESTAMP," +
+                "    a_binary BINARY," +
+                "    a_string_array ARRAY<STRING>," +
+                "    a_struct_array ARRAY<STRUCT<a_string: STRING>>," +
+                "    a_map MAP<STRING, STRING>," +
+                "    a_complex_map MAP<STRING, STRUCT<a_string: STRING>>," +
+                "    a_struct STRUCT<a_string: STRING, a_int: INT>," +
+                "    a_complex_struct STRUCT<nested_struct: STRUCT<a_string: STRING>, a_int: INT>" +
```

Comment:
Specifying all types in **_a_** table reduces the readability. Is testing with data provider per type is too slow?

Reply:
Do note that we have 3 column mappings x 2 (partitioned/unpartitioned) already.
I opted intentionally in this case to trade test runtime for readability.